src/common.speech/ConversationTranslator/ConversationConnection.ts

```diff
@@ -0,0 +1,460 @@
+
+        // clear the temp subscription key if it's a participant joining
+        if (speechConfig.getProperty(PropertyId[PropertyId.SpeechServiceConnection_Key]) === "abcdefghijklmnopqrstuvwxyz012345") {
+            speechConfig.setProperty(PropertyId[PropertyId.SpeechServiceConnection_Key], "");
+        }
+
+        this.privSpeechTranslationConfig = speechConfig;
+        this.privIsConversationConnected = false;
+        this.privParticipants = new InternalParticipants();
+        this.privIsReady = false;
+        this.privIsExiting = false;
+    }
+
+    public canceled: (sender: ConversationConnection, event: ConversationTranslationCanceledEventArgs) => void;
+    public conversationExpiration: (sender: ConversationConnection, event: ConversationExpirationEventArgs) => void;
+    public participantsChanged: (sender: ConversationConnection, event: ConversationParticipantsChangedEventArgs) => void;
+    public sessionStarted: (sender: ConversationConnection, event: SessionEventArgs) => void;
+    public sessionStopped: (sender: ConversationConnection, event: SessionEventArgs) => void;
+    public textMessageReceived: (sender: ConversationConnection, event: ConversationTranslationEventArgs) => void;
+    public transcribed: (sender: ConversationConnection, event: ConversationTranslationEventArgs) => void;
+    public transcribing: (sender: ConversationConnection, event: ConversationTranslationEventArgs) => void;
+
+    public connect(room: IInternalConversation): void {
+
+        try {
+
+            this.privRoom = room;
+            this.privParticipants.meId = room.participantId;
+
+            if (!!this.privConversationRecognizer) {
+                // close the existing recognizer
+                this.privConversationRecognizer.disconnect();
+                this.privConversationRecognizer = undefined;
+            }
+
+            this.privConversationRecognizer = new ConversationTranslatorRecognizer(this.privSpeechTranslationConfig, undefined);
+
+            this.privRecoConnection = Connection.fromRecognizer(this.privConversationRecognizer);
+
+            this.privRecoConnection.connected = (e: ConnectionEventArgs) => {
+                // tslint:disable-next-line: no-console
+                // console.log("connection connected", e);
+                this.privIsConnected = true;
+            };
+            this.privRecoConnection.disconnected = (e: ConnectionEventArgs) => {
+                // tslint:disable-next-line: no-console
+                // console.log("connection disconnected", e);
+                this.privIsConnected = false;
+                if (this.privIsExiting) {
+                    this.cancel();
+                }
+            };
+
+            this.privConversationRecognizer.connectionOpened = ((r: ConversationTranslatorRecognizer, e: SessionEventArgs) => {
+
+                this.privIsConnected = true;
+
+                try {
+                    if (!!this.sessionStarted) {
```

Comment:
You should be sending SessionStarted and SessionStopped events based on the recognizer state. Keep in mind the fact that there are two web socket connections is really an implementation detail. Another point is that having a speech recognition connection is optional too

From the end user point of view, they should get a SessionStarted event whenever the overall "conversation" starts (e.g. after join right now), and SessionStopped whenever the "conversation" state is terminal e.g. we deleted the conversation.